demos/device_defender_for_aws/defender_demo.c

```diff
@@ -374,7 +392,9 @@ static BaseType_t collectDeviceMetrics( void )
 {
     BaseType_t status = pdFAIL;
     MetricsCollectorStatus_t metricsCollectorStatus;
-    uint32_t numOpenTcpPorts, numOpenUdpPorts, numEstablishedConnections;
+    uint32_t numOpenTcpPorts, numOpenUdpPorts, numEstablishedConnections, i;
+    UBaseType_t tasksWritten = { 0 };
```

Comment:
`UBaseType_t` is just an integer type, not an aggregator type.
```suggestion
    UBaseType_t tasksWritten = 0;
```